edgexfoundry/device-sdk-go
# feat: support YAML format for ProvisionWatcher definition file

closes #1381







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

1. Run `device-simple` from this branch
2. Verify bootstrap logs includes something similiar:
```shell
level=INFO ts=2023-03-15T14:02:31.201989Z app=device-simple source=provisionwatchers.go:47 msg="Loading pre-defined provision watchers from /Users/chris/Workspace/EdgeX/device-sdk-go/example/cmd/device-simple/res/provisionwatchers(2 files found)"
level=INFO ts=2023-03-15T14:02:31.202598Z app=device-simple source=provisionwatchers.go:84 msg="ProvisionWatcher Simple-Provision-Watcher not found in Metadata, adding it..."
level=DEBUG ts=2023-03-15T14:02:31.206317Z app=device-simple source=callback.go:58 msg="System event received on message queue. Topic: edgex/system-events/core-metadata/provisionwatcher/add/device-simple/Simple-Device, Correlation-id: 3da7e62b-95a8-481c-81b1-0cc715b22cba"
level=DEBUG ts=2023-03-15T14:02:31.207866Z app=device-simple source=callback.go:202 msg="provision watcher Simple-Provision-Watcher added"
```

## New Dependency Instructions (If applicable)
